docs/cli/app-types.md

```diff
@@ -0,0 +1,141 @@
+a certain execution model or are based on what the application can "do"; examples of these are "console application", 
+"web application" etc. All of these types of applications (and more) can be created with .NET Core, since it is 
+a general purpose development platform.
+
+However, given its unique nature, .NET Core also has another axis on which to observe the type of the application 
+and that is application's *portability*. Portability here means, essentially, where can you run your application and what are the 
+pre-requisites that you need to satisfy in order for your application to run on a given machine. This document 
+deals with this axis, portability, and outlines the two main types of portability that .NET Core enables. 
+
+There are two main types that we can observe: 
+
+1. Portable application
+    * As a sub-category of this we have the portable application with native dependencies
+2. Self-contained application
+
+## Portable application
+Fully portable application is an IL application that targets a shared component called **the shared runtime**. 
+This makes it portable to any platform (operating system) that the shared runtime runs on. 
+
+This type of application will only carry its own code and dependencies that are outside of the shared runtime. 
+As an example, let's say you are making a console application that has the ability to invoke a certain REST API 
+and deserialize the returned JSON into a type and then display it. You have everything you need for this small 
+application expect for a good JSON parser; for this, you add a dependency to your `project.json` to include 
+[Json.NET](https://www.nuget.org/packages/Newtonsoft.Json/). Once you publish your application using `dotnet publish`, 
+you will see that only your application's code and JSON.net have been published in the output. 
+The shared runtime remains outside of your application's dependency closure. 
+
+Of course, this type of application requires you to have a shared runtime installed. Otherwise, you will not be able 
+to run the application. 
+
+In order to create a portable application, the only thing you need is to target the shared runtime in your `project.json`
+and of course have your frameworks aligned as the below sample shows. 
+
+```json
+"dependencies": {
+    "Microsoft.NETCore.App": {  
+        "version": "1.0.0-*",
+        "type": "platform"
+    }
+},
+"frameworks": {
+    "netcoreapp1.0": {}
+}
+```
+
+### Portable application with native dependencies
+A sub-group of the above, this type is a portable application that has native dependencies specified 
+somewhere in its dependency chain. This application is as portable as all of its native dependencies 
+are portable. You will be able to run the application on any platform that your native dependencies can 
+run on. Prime example of this is Kestrel, the ASP.NET Core cross-platform web server. It is built on top of 
+`libuv` which is its native dependency. 
```

Comment:
do we need to explain what libuv is or point to a place where users can find more info? I had to look it up